src/features/authorSlice.ts

```diff
@@ -0,0 +1,63 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { getUsers } from '../api/users';
+import { Post } from '../types/Post';
+
+type UsersState = {
+  users: User[];
+  isLoading: boolean;
+  errorMessage: string;
+  selectedUser: User | null;
+  setSelectedPost: Post | null;
+};
+
+const initialState: UsersState = {
+  users: [],
+  isLoading: false,
+  errorMessage: '',
+  selectedUser: null,
+  setSelectedPost: null,
+};
+
+export const init = createAsyncThunk('users/fetch', () => {
+  return getUsers();
+});
+
+const authorSlice = createSlice({
+  name: 'users',
+  initialState,
+  reducers: {
+    setUser: (state, action) => {
+      // eslint-disable-next-line no-param-reassign
```

Comment:
Disable this rule for the whole file to not repeat it several times

Reply:
Thanks I didn't think of that